rules/PoiGeneric.js

```diff
@@ -164,22 +173,34 @@ function additiveScore(map, e1, e2) {
 
     var reason = result.reasons;
 
-    var searchRadius = Math.max(exports.getSearchRadius(e1), 
-        exports.getSearchRadius(e2));
+    var t1 = e1.getTags().toDict();
+    var t2 = e2.getTags().toDict();
+
+    // if there is no type information to compare the name becomes more
+    // important
+    var oneGeneric = hasTypeTag(e1) == false || hasTypeTag(e2) == false;
+
+    var e1SearchRadius = exports.getSearchRadius(e1);
+    var e2SearchRadius = exports.getSearchRadius(e2);
+    var searchRadius;
+    if (oneGeneric)
+    {
+      searchRadius = Math.max(e1SearchRadius, e2SearchRadius);
+    }
+    else
+    {
+      searchRadius = Math.min(e1SearchRadius, e2SearchRadius);
+    }
 
     var d = distance(e1, e2);
 
     if (d > searchRadius) {
+        hoot.debug(
+          "distance: " + d + " greater than search radius: " + searchRadius + "; returning score: " +
+          result.score);
```

Comment:
Unlike the C++ debug statements the Node.js statements will compose the string every time and then filter the debug statement out before it is printed. Not necessarily a huge deal, but string manipulations aren't particularly fast.